nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1048,6 +1049,7 @@ def build_pretraining_data_loader(
                     global_batch_size=self.cfg.global_batch_size,
                     rampup_batch_size=self.cfg.get('rampup_batch_size', None),
                     pad_samples_to_global_batch_size=pad_samples_to_global_batch_size,
+                    compute_length_in_micro_batches=(dataset_type == "validation"),
```

Comment:
Indeed I share the same concern (see my point above "Changing the length of the validation dataloader might have unintended consequences") => that's one reason I was asking for feedback, in case someone would tell me "no don't do that this is going to break xyz".

If you agree this looks like a good idea, I'll finish this PR and will run the tests to see if anything breaks...

Reply:
Right..Yes I think it's a good idea to check the tests if it's breaking anything.

Reply:
Adding @ericharper in case he has any insights here.